File: benchmarks/src/perl/exception_ladder.pl
use Time::HiRes qw(time);
sub f3 {my$i=shift;die"boom" if$i%7==0;$i*2}
sub f2 {f3(@_)}
sub f1 {f2(@_)}
my$N=200000;my($acc,$caught)=(0,0);my$start=time();
for my$i(1..$N){eval{$acc=($acc+f1($i))%1000003};$caught++ if$@}
my$end=time();my$ms=($end-$start)*1000;print"Result: ${acc}_${caught}\n";printf"Time: %.0fms\n",$ms;

File: benchmarks/src/python/exception_ladder.py
import time

def f3(i):
    if i % 7 == 0:
        raise Exception("boom")
    return i * 2

def f2(i):
    return f3(i)

def f1(i):
    return f2(i)

N = 200000
acc = 0
caught = 0
start = time.time() * 1000
for i in range(1, N + 1):
    try:
        acc = (acc + f1(i)) % 1000003
    except Exception:
        caught += 1
end = time.time() * 1000

print(f"Result: {acc}_{caught}")
print(f"Time: {int(end - start)}ms")
